pkg/collector/check/helpers.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package check
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+)
+
+// GetInfo returns information about a specific check instances. If 'includeConfig' it true, the instance configuration
+// will be scrubbed and included in the returned map
+func GetInfo(c Info, includeConfig bool) map[string]interface{} {
```

Comment:
I understand the idea, but the fact that the return value is tightly coupled to what the backend expects in the inventory payload makes it very non-generic in my eyes.

Besides, it is not clear that a second use case will ever come by, much less that it will be able to use the same data format.

Reply:
Let's call it `GetMetadata` and `metadata.go` then. WDYT?